compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -908,6 +908,62 @@ TopkOp::reifyResultShapes(OpBuilder &b,
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+//===----------------------------------------------------------------------===//
+// ArgmaxOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult ArgmaxOp::verify() {
+  Operation *op = getOperation();
+
+  // Check number of inputs and outputs.
+  if (getNumDpsInputs() != 1) {
+    return op->emitOpError("expected exactly one input operand (values)");
+  }
+
+  if (getNumDpsInits() != 2) {
+    return op->emitOpError("expected two output operands (value and index)");
+  }
+
+  uint64_t dim = getDimension();
+  if (dim >= getInputRank()) {
+    return op->emitOpError("reduction dimension exceeds input rank");
+  }
+
+  ShapedType inputType = getInputType();
+  ShapedType outputValueType = cast<ShapedType>(outputValue().getType());
+  ShapedType outputIndexType = cast<ShapedType>(outputIndex().getType());
```

Comment:
```suggestion
  auto outputValueType = cast<ShapedType>(outputValue().getType());
  auto outputIndexType = cast<ShapedType>(outputIndex().getType());
```
See https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable